apps/prairielearn/src/models/user.ts

```diff
@@ -38,6 +52,38 @@ export async function selectOrInsertUserByUid(uid: string): Promise<User> {
   return await queryRow(sql.select_or_insert_user_by_uid, { uid }, UserSchema);
 }
 
+export async function insertUserLti({
+  uid,
+  name,
+  lti_course_instance_id,
+  lti_user_id,
+  lti_context_id,
+  institution_id,
+}: {
+  uid: string;
+  name: string | null;
+  lti_course_instance_id: string;
+  lti_user_id: string;
+  lti_context_id: string;
+  institution_id: string;
+}): Promise<User> {
+  return await queryRow(
+    sql.insert_user_lti,
+    { uid, name, lti_course_instance_id, lti_user_id, lti_context_id, institution_id },
+    UserSchema,
+  );
+}
+
+export async function updateUserName({
+  user_id,
+  name,
+}: {
+  user_id: string;
+  name: string;
+}): Promise<User> {
+  return await queryRow(sql.update_user_name, { user_id, name }, UserSchema);
```

Comment:
In PT, we conventionally insert audit events inside functions like these, rather than leaving it to the caller. This admittedly doesn't matter much yet as `updateUserName` is only called from one place. This is not a blocker.

Reply:
I made this change in dd1fb84561aad1c2081431115ef944e79e0558ff